### ISSUE TITLE
Add-on fails with "'external_hostname' is empty"

### ISSUE THREAD

**Issue by dschlegel** [brenner-tobias/addon-cloudflared#135]
### The problem

OS: Ubuntu 22.04 LTS
Home Assistant: 2022.7.7
Docker: 20.10.17

Whenever I try to start the Add-on it fails with the above mentioned error message:
`FATAL: 'external_hostname' is empty, please enter a valid String`

The folder mounted as `/data` contains a valid `options.json` config:
```json
{
  "additional_hosts": [],
  "external_hostname": "test.com",
  "tunnel_name": "homeassistant",
  "tunnel_token": "",
  "data_folder": "config",
  "custom_config": false,
  "log_level": "trace"
}
```

### What version of Cloudflared has the issue?

2.0.5

### What type of installation are you running?

Home Assistant Supervised

### Add-on YAML Configuration

```yaml
additional_hosts: []
external_hostname: test.com
tunnel_name: homeassistant
tunnel_token: ''
data_folder: config
custom_config: false
log_level: trace
```

### Anything in the logs that might be useful for us?

```txt
s6-rc: info: service s6rc-oneshot-runner: starting
s6-rc: info: service s6rc-oneshot-runner successfully started
s6-rc: info: service fix-attrs: starting
s6-rc: info: service fix-attrs successfully started
s6-rc: info: service legacy-cont-init: starting
cont-init: info: running /etc/cont-init.d/00-banner.sh
curl: (7) Couldn't connect to server
[09:51:41] ERROR: Something went wrong contacting the API
cont-init: info: /etc/cont-init.d/00-banner.sh exited 0
cont-init: info: running /etc/cont-init.d/01-log-level.sh
curl: (7) Couldn't connect to server
[09:51:41] ERROR: Something went wrong contacting the API
cont-init: info: /etc/cont-init.d/01-log-level.sh exited 0
s6-rc: info: service legacy-cont-init successfully started
s6-rc: info: service init-cloudflared-log: starting
s6-rc: info: service init-cloudflared-log successfully started
s6-rc: info: service init-cloudflared-config: starting
[09:51:42] INFO: Checking Add-on config...
[09:51:42] FATAL: 'external_hostname' is empty, please enter a valid String
s6-rc: warning: unable to start service init-cloudflared-config: command exited 1
/run/s6/basedir/scripts/rc.init: warning: s6-rc failed to properly bring all the services up! Check your logs (in /run/uncaught-logs/current if you have in-container logging) for more information.
/run/s6/basedir/scripts/rc.init: fatal: stopping the container.
s6-rc: info: service init-cloudflared-log: stopping
s6-rc: info: service init-cloudflared-log successfully stopped
s6-rc: info: service legacy-cont-init: stopping
s6-rc: info: service legacy-cont-init successfully stopped
s6-rc: info: service fix-attrs: stopping
s6-rc: info: service fix-attrs successfully stopped
s6-rc: info: service s6rc-oneshot-runner: stopping
s6-rc: info: service s6rc-oneshot-runner successfully stopped
```

### Steps to reproduce the issue

Start the Add-on.

### Additional information

I don't fully understand how the initialization process is working but maybe this error is related to other errors shown in the log:
`curl: (7) Couldn't connect to server`

Usually the supervisor entry is passed as an extra host docker and added to `/etc/hosts`. After creating the container and before starting it the file looks like this:
```
127.0.0.1       localhost
::1     localhost ip6-localhost ip6-loopback
fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
172.30.32.2     hassio
172.30.32.2     supervisor
172.17.0.6      9074a9fa-cloudflared.local.hass.io 9074a9fa-cloudflared
```

After starting the Add-on it loogs like this:
```
127.0.0.1       localhost
::1     localhost ip6-localhost ip6-loopback
fe00::0 ip6-localnet
ff00::0 ip6-mcastprefix
ff02::1 ip6-allnodes
ff02::2 ip6-allrouters
172.30.32.2     hassio
172.30.33.1     9074a9fa-cloudflared.local.hass.io 9074a9fa-cloudflared
```

Somehow the `supervisor` entry was removed from the list. Is is possible that this is causing the issue?

**Comment by elcajon** [comment]
Hi Daniel, thanks for your research on this. 
There are already some related issues e.g. #111 or #122. 
I am pretty sure that this is caused by the host OS (Ubuntu) which [isn't officially supported](https://github.com/home-assistant/architecture/blob/master/adr/0014-home-assistant-supervised.md#supported-operating-system-system-dependencies-and-versions) by HA Supervised deployment method. 

As you have already took a closer look at this, do you expect any similar issues with other add-ons related on the same [base image](https://github.com/hassio-addons/addon-base), e.g. [addon-ssh](https://github.com/hassio-addons/addon-ssh)?
If so, I think there isn't anything we can do from add-on perspective to fix this.

To directly answer your question. Yes, the add-on contacts the supervisor to get add-on information. Somehow this connection is not possible.

Best, Max

**Comment by dschlegel** [comment]
Hi Max,
I also think that it is caused by the OS. addon-ssh is working without a problem but that might be a coincidence.

After some more testing I fount out that it is related to the supervisor. If I manually start the container on the default docker bridge it is able to connect to the supervisor but if I start it on the `hassio` bridge it fails with an `invalid token` error.

I will investigate further but I also don't think that this issue is on your side. 

Regards,
Daniel

**Comment by elcajon** [comment]
I just quickly created an Ubuntu VM to look at the problem. Since the ssh-addon was working I thought it must have something to do with our add-on. In fact, the Apparmor profile is causing the problem on Ubuntu. 

Since I (still) have no idea about Apparmor, I would recommend deleting the profile and accepting the security rating of 6 (instead 7). What do you think @brenner-tobias ?

**Comment by elcajon** [comment]
@dschlegel if you clone this repo to /addons and remove the apparmor.txt file, the add-on should work as expected.

**Comment by dschlegel** [comment]
@elcajon Thank you very much. It is now working without a problem!

**Comment by RaveGun** [comment]
> @dschlegel if you clone this repo to /addons and remove the apparmor.txt file, the add-on should work as expected.

I have the same Ubuntu + Supervised combination.
Can you please elaborate on this?

I went and deleted the apparmor.txt out of the add on install location:
/usr/share/hassio/addons/git/9074a9fa/cloudflared#
But it did not helped.

Thank you.

**Comment by dschlegel** [comment]
@RaveGun you need to add this repository as a local addon.

1) Remove the old cloudflared addon
1) Clone this repository into `/usr/share/hassio/addons/local` 
1) Delete the `cloudflared/apparmor.txt` file
1) I also needed to change the architecture to aarch64 in `cloudflared/Dockerfile` (lines 1 and 9) and change the `yq` version in the last line of `cloudflared/rootfs/build.sh` to `yq=4.25.1-r3`
1) Restart HomeAssistant
1) Now you should have a `Local add-ons` section in the store which contains the cloudflared addon

```bash
# Clone the repository
cd /usr/share/hassio/addons/local
git clone https://github.com/brenner-tobias/addon-cloudflared
```

**Comment by RaveGun** [comment]
Thank you very much.
It is working now.

**Comment by elcajon** [comment]
Should be fixed in v2.0.6 (just released, maybe you need to check for updates in HA first)
